src/index.ts

```diff
@@ -543,15 +599,16 @@ function propertyConditions(
   path: string,
   arrayDepth: number,
   records: readonly IRecord[],
-  options: IProcessOptions
+  options: IProcessOptions,
+  calledByErrorOut: boolean
 ): string | null {
   const { debug } = options
   const propertyName = property.name
 
   const isIdentifier = propertyName[0] !== '"'
   const varName = isIdentifier
-    ? `${objName}.${propertyName}`
-    : `${objName}[${propertyName}]`
+    ? `${objName}?.${propertyName}`
```

Comment:
This `?` shouldn't be necessary because if the object is `undefined` then it should be checked _before_ the code attempts to access a property.